firedrake/extrusion_utils.py

```diff
@@ -58,7 +58,7 @@ def make_extruded_coords(extruded_topology, base_coords, ext_coords,
                 ext_coords[2*d+1][%(base_coord_dim)d] = *layer_height * (layer[0][0] + 1);
             }
         }""" % {'base_map_arity': base_coords.cell_node_map().arity,
-                'base_coord_dim': base_coords.function_space().dim},
+                'base_coord_dim': base_coords.function_space().shape_dim},
```

Comment:
`.ufl_element().value_size()` is not quite the same as you mentioned. What I wanted to suggest is that `value_size` might be a more consistent/familiar name than `shape_dim`.